content/docs/faq-state.md

```diff
@@ -31,38 +33,74 @@ incrementCount() {
 }
 
 handleSomething() {
-  // this.state.count is 1, then we do this:
+  // Let's say `this.state.count` starts at 0.
+  this.incrementCount();
+  this.incrementCount();
   this.incrementCount();
-  this.incrementCount(); // state wasn't updated yet, so this sets 2 not 3
+  // When React re-renders the component, `this.state.count` will be 1, but you expected 3.
+
+  // This is because `incrementCount()` function above reads from `this.state.count`,
+  // but React doesn't update `this.state.count` until the component is re-rendered.
+  // So `incrementCount()` ends up reading `this.state.count` as 0 every time, and sets it to 1.
+
+  // The fix is described below!
 }
 ```
 
 See below for how to fix this problem.
 
 ### How do I update state with values that depend on the current state? 
 
-Pass a function instead of an object to setState to ensure the call always uses the most updated version of state (see below). 
+Pass a function instead of an object to `setState` to ensure the call always uses the most updated version of state (see below). 
 
-### What is the difference between passing an object or a function in setState?
+### What is the difference between passing an object or a function in `setState`?
 
 Passing an update function allows you to access the current state value inside the updater. Since `setState` calls are batched, this lets you chain updates and ensure they build on top of each other instead of conflicting:
 
 ```jsx
 incrementCount() {
   this.setState((prevState) => {
+    // Important: read `prevState` instead of `this.state` when updating.
     return {count: prevState.count + 1}
   });
 }
 
 handleSomething() {
-  // this.state.count is 1, then we do this:
+  // Let's say `this.state.count` starts at 0.
   this.incrementCount();
-  this.incrementCount(); // count is now 3
+  this.incrementCount();
+  this.incrementCount();
+
+  // If you read `this.state.count` now, it would still be 0.
+  // But when React re-renders the component, it will be 3.
+  
+  // If you need to know the pending state now, keep it in a local variable.
```

Comment:
How about an example of doing this?

```js
let newCount

this.setState(prevState => {
  const oldCount = prevState.count
  newCount = oldCount + 1
  return  { count: newCount }
})
```

Reply:
Ah, merged. Will make a PR.

Reply:
This particular example won't work because the updater might be applied later. I don't mind having an example but it'll probably need to be its own subsection.